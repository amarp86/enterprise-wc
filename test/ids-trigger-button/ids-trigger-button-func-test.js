/**
 * @jest-environment jsdom
 */
import IdsTriggerButton from '../../src/ids-trigger-field/ids-trigger-button';

describe('IdsTriggerButton Component', () => {
  let triggerButton;

  beforeEach(async () => {
    const elem = new IdsTriggerButton();
    document.body.appendChild(elem);
    triggerButton = document.querySelector('ids-trigger-button');
  });

  afterEach(async () => {
    document.body.innerHTML = '';
  });

  it('renders with no errors', () => {
    const errors = jest.spyOn(global.console, 'error');
    const elem = new IdsTriggerButton();
    document.body.appendChild(elem);
    elem.remove();
    expect(document.querySelectorAll('ids-trigger-button').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    expect(triggerButton.outerHTML).toMatchSnapshot();
  });

  it('defaults tabbable to true', () => {
    expect(triggerButton.tabbable).toEqual(true);
    triggerButton.removeAttribute('tabbable');
    expect(triggerButton.tabbable).toEqual(true);
  });

  it('supports tabbable', () => {
    triggerButton.tabbable = true;

    expect(triggerButton.shadowRoot.querySelector('button').getAttribute('tabindex')).toEqual('0');
    expect(triggerButton.tabbable).toEqual('true');

    triggerButton.tabbable = false;

    expect(triggerButton.shadowRoot.querySelector('button').getAttribute('tabindex')).toEqual('-1');
    expect(triggerButton.tabbable).toEqual('false');
  });
});
