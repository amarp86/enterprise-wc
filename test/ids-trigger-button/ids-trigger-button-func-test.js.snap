// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsTriggerButton Component renders correctly 1`] = `"<ids-trigger-button></ids-trigger-button>"`;
